### DISCUSSION TITLE
No conditional rendering of Disclosures at all?

### DISCUSSION THREAD

**Discussion by hibachrach** [ariakit/ariakit#847] [Questions] [answered]
The docs say

> You shouldn't conditionally render the `DisclosureContent` component as this will make some Reakit features not work. Instead, you can use render props and conditionally render the underneath element:

Does that mean that if I had, say, a SPA and one of the pages had a disclosure, that it would not work properly? Or just that `Disclosure`s and corresponding `DisclosureContent`s have to be rendered "in sync"?

**Comment by diegohaz** [comment, marked as answer]
> Or just that Disclosures and corresponding DisclosureContents have to be rendered "in sync"?

`Disclosure`, `DisclosureContent` and `useDisclosureState` should be sync, but you can always use render props as the docs say or put them in a separate component and render that component asynchronously. There's no problem in using it in SPAs.

```jsx
function AnotherComponent() {
  const state = useDisclosureState();
  return (
    <div>
      <Disclosure {...state} />
      <DisclosureContent {...state} />
    </div>
  );
}

{show && <AnotherComponent>}
```
